### PR TITLE
Support PROs of contentype Text

### LINKED ISSUE #3430: Only show PROs of contentType "Text"
**Is your feature request related to a problem? Please describe.**

We accidentally allowed all types of PRO
We should also have checked the "contentType"!

**Describe the solution you'd like**
Add a second whitelist to the content type preprocessor rule. This should be compared against the `contentType` key in `PRO` assets.

Only add `"Text"` to that second whitelist (for now).

**Describe alternatives you've considered**
nope

**Testing notes**
[Tester to complete]

Dev insight: Will Cypress tests be required or are unit tests sufficient? Will there be any potential regression? etc

- [ ] This feature is expected to need manual testing.

**Additional context**

### PR DESCRIPTION
**Pull request by DenisHdz**
Resolves #3430

**Overall change:** 
Display only standard (Text) link promos (PROs).

**Code changes:**
- Add a second whitelist to the content type preprocessor rule to filter PRO assets of content type `Text`.
- Add a new test to filter also by `contentType`

---

- [X] I have assigned myself to this PR and the corresponding issues
- [X] Automated (jest and/or cypress) tests added (for new features) or updated (for existing features)
- [X] This PR requires manual testing

### CODE REVIEW

File: src/app/lib/utilities/preprocessor/rules/filterContentType.js [hunk 1]
```diff
@@ -17,7 +18,10 @@ const filterUnknownContentTypes = data => {
       newGroup.items = group.items.filter(
         item =>
           (item.assetTypeCode || item.cpsType) &&
-          whitelist.includes(item.cpsType || item.assetTypeCode),
+          whitelist.includes(item.cpsType || item.assetTypeCode) &&
+          (item.assetTypeCode === 'PRO'
+            ? !!contentTypes.includes(item.contentType)
+            : true),
```
Comment (andrew-nowak) [inline src/app/lib/utilities/preprocessor/rules/filterContentType.js]: this ternary can simplify to
`(item.assetTypeCode !== 'PRO' || contentTypes.includes(item.contentType))`

(Array.includes already returns a boolean, you don't need to `!!` it)

**Review by andrew-nowak — APPROVED** [review]
logic works & tested locally, just a possible simplification

File: src/app/lib/utilities/preprocessor/rules/filterContentType.js (hunk 1 above)
Comment (ghost) [reply src/app/lib/utilities/preprocessor/rules/filterContentType.js]: Could we also simplify the mapping of `group` and filtering of `group.items`. Its getting hard to follow.

File: src/app/lib/utilities/preprocessor/rules/filterContentType.js [hunk 2]
```diff
@@ -14,11 +15,17 @@ const filterUnknownContentTypes = data => {
...
     if (Array.isArray(group.items)) {
-      newGroup.items = group.items.filter(
-        item =>
-          (item.assetTypeCode || item.cpsType) &&
-          whitelist.includes(item.cpsType || item.assetTypeCode),
-      );
+      newGroup.items = group.items.filter(item => {
+        const itemType = item.assetTypeCode || item.cpsType;
+        const validItemType = whitelist.includes(
+          item.cpsType || item.assetTypeCode,
```
Comment (ghost) [inline src/app/lib/utilities/preprocessor/rules/filterContentType.js]:
```suggestion
         itemType
```

File: src/app/lib/utilities/preprocessor/rules/filterContentType.js (hunk 1 above)
Comment (DenisHdz) [reply src/app/lib/utilities/preprocessor/rules/filterContentType.js]: Done!

**Review by ghost — APPROVED** [review]

**Review by dr3 — APPROVED** [review]

**Comment by david-boydell** [comment]
Sweet, this is ready for merging.
